### DISCUSSION TITLE
GigaChain Agents Module 422 response error

### DISCUSSION THREAD

**Discussion by Niek352** [ai-forever/gigachain#71] [Q&A]
### Checked other resources

- [X] I added a very descriptive title to this question.
- [X] I searched the LangChain documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
giga = GigaChat(credentials='xxx', scope="GIGACHAT_API_PERS")

tools = [nTool1, nTool2]
#Если создать агента через ZeroShootAgent то ошибок нету, но он не поддерживает много тулов
agent = create_gigachat_functions_agent(giga, tools)

agent_executor = AgentExecutor(
    agent=agent,
    tools=tools,
    verbose=True,
)

#Тут при вызове g

output = agent_executor.invoke(
        {
            "input": "Скачай из рабочей папки файл картинку: XXX, и покажи его имя"
        }
    )
```

### Description

При попытке иновкнуть агента прилетает ошибка с сервера.

`raise ResponseError(response.url, response.status_code, b"", response.headers)
gigachat.exceptions.ResponseError: (URL('https://gigachat.devices.sberbank.ru/api/v1/chat/completions'), 422, b'', Headers([('server', 'nginx'), ('date', 'Mon, 26 Feb 2024 09:21:15 GMT'), ('content-type', 'application/json; charset=utf-8'), ('content-length', '70'), ('connection', 'keep-alive'), ('access-control-allow-credentials', 'true'), ('access-control-allow-headers', 'Origin, X-Requested-With, Content-Type, Accept, Authorization'), ('access-control-allow-methods', 'GET, POST, DELETE, OPTIONS'), ('access-control-allow-origin', 'https://beta.saluteai.sberdevices.ru'), ('x-request-id', 'bb509e4f-589b-4e34-937c-809dd2395bfd'), ('x-session-id', '0c5501d8-1be9-411e-b3cd-c65edb3b90e0'), ('allow', 'GET, POST'), ('strict-transport-security', 'max-age=31536000; includeSubDomains'), ('allow', 'GET, POST'), ('strict-transport-security', 'max-age=31536000; includeSubDomains')]))
`

### System Info

Библиотеки:
gigachain
googleServices -> Google Drive Api

**Comment by urfv** [comment]
@Rai220 отвечал [тут](https://giter.club/ai-forever/gigachain/issues/69):

> Здравствуйте! Нужен доступ именно к "модели с поддержкой функций", сейчас она есть только у ограниченного круга лиц в рамках тестирования. В ближайшее время будет доступна для всех. Если очень нужна до запуска, можно обратиться [gigachat@sberbank.ru](mailto:gigachat@sberbank.ru), вероятно помогут получить к ней доступ (но тут я не могу гарантировать).
